src/stereotype_client.js

```diff
@@ -507,28 +580,44 @@ class StereotypeClient {
    * in case the preference for async execution was respected or `201` in case the server
    * decided to ignore the preference and execute the request synchronously.
    *
-   * @param {string} idTemplate
+   * @param {string} templateUrl
    * @param {object} propertyBag A JSON object that contains the data to be populated in the template.
    * @param {boolean} getMaterializationId Return the materialization id instead of the materialization
    *    body. We can use that id later to fetch the materialized template without resending the properties.
    *    Defaults to false.
    */
-  materializeAsync(idTemplate, propertyBag, getMaterializationId = false, skipCache = false) {
-    return this._materialize(idTemplate, propertyBag, getMaterializationId, true, skipCache);
+  materializeAsync(templateUrl, propertyBag, getMaterializationId = false, skipCache = false) {
+    return this._materialize(templateUrl, propertyBag, getMaterializationId, true, skipCache);
   }
 
   _getUrl(path) {
     return this.baseUrl + path;
   }
 
-  _materialize(idTemplate, propertyBag, getMaterializationId = false, preferAsync = false, skipCache = false) {
+  _verifyTemplateUrl(path, templateUrl) {
+    const parts = templateUrl.split('/');
+    if (parts.length <= 0) {
```

Comment:
A split never returns less than zero elements. At worst you can get a zero-element array: ''.split("/").